src/accelerate/accelerator.py

```diff
@@ -1916,12 +1922,12 @@ def _prepare_ipex(self, *args):
                 optimizer = obj
         if optimizer is not None and model is not None:
             dtype = torch.bfloat16 if self.state.mixed_precision == "bf16" else None
-            if self.device.type == "xpu" and is_xpu_available():
+            if self.device.type == "xpu":
                 model = model.to(self.device)
-                model, optimizer = torch.xpu.optimize(
-                    model, optimizer=optimizer, dtype=dtype, inplace=True, level="O1"
-                )
-            else:
+            # ipex.optimize() is available only for IPEX, both IPEX-CPU and IPEX-XPU
+            if is_ipex_available():
```

Comment:
Maybe change that by `self.state.use_ipex` (or add at least)?

Reply:
I afraid this might break IPEX-XPU path.

Note that `use_ipex` is currently used with CPU path to differentiate the case when IPEX-CPU optimization should or should not be used:
https://github.com/huggingface/accelerate/blob/c0faec766c02973edb5b75b97fee5fe4ee1a7c69/src/accelerate/accelerator.py#L1288
In case of IPEX-XPU this flag was not used and I am not sure whether it will be `=True`.

Reply:
Sound good !